core/commands/destroy_command.rb

```diff
@@ -166,4 +191,45 @@ def execute
     end
     SUCCESS_RESULT
   end
+
+  private
+
+  def remember_aws_instance_id(configuration, node)
+    return if configuration.provider != 'aws'
+
+    @aws_instance_ids = {}
+    if node.empty?
+      configuration.node_names.each do |node_name|
+        @aws_instance_ids[node_name] = get_aws_instance_id_by_node_name(configuration, node_name)
+      end
+    else
+      @aws_instance_ids[node] = get_aws_instance_id_by_node_name(configuration, node)
+    end
+  end
+
+  def get_aws_instance_id_by_node_name(configuration, node_name)
+    aws_instance_id_path = "#{configuration.path}/.vagrant/machines/#{node_name}/aws/id"
+    return nil unless File.file?(aws_instance_id_path)
+    return File.read(aws_instance_id_path)
+  end
+
+  def get_aws_instance_state_by_id(instance_id)
+    return nil if instance_id.nil?
+    instances_description = run_command_and_log('aws ec2 describe-instances --profile mdbci')[:output]
+    instances_array = JSON.parse(instances_description).to_h
+
+    instance = nil
+    instances_array['Reservations'].each do |reservation|
```

Comment:
Use [find](http://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-find) iterator if you need to find something in the array.